aws/ec2.go

```diff
@@ -160,6 +160,10 @@ func getSubnets(svc ec2iface.EC2API, vpcID string) ([]*subnetDetails, error) {
 					aws.String(vpcID),
 				},
 			},
+                        {
+                                Name:   aws.String("tag:" + clusterIDTagPrefix + clusterId),
+                                Values: []*string{aws.String(resourceLifecycleOwned)},
```

Comment:
This is a bit too strict for us as we share subnets with the tag `kubernetes.io/cluster/<id>=shared`.

We could add `shared` as another possible value, but to be backwards compatible we should also support the current case were no cluster id is defined on the subnets.

Reply:
Perhaps it's better to just look for the tag key vs. using the filter. That's what the code this logic was based on is doing, this reference is just above the block referenced in the comments of the code:

https://github.com/kubernetes/kubernetes/blob/65efeee64f772e0f38037e91a677138a335a7570/pkg/cloudprovider/providers/aws/aws.go#L2923-L2928

I will have a look at using that approach instead.

Reply:
@jhohertz yup, that makes sense, also because the logic then is the same for Kubernetes services of type load balancer, which is why we modeled the logic after that (but obviously didn't go the full way as you found).

Thanks for the effort!